File: backend/project-service/src/main/java/com/buildflow/project/service/ProjectService.java
package com.buildflow.project.service;

import com.buildflow.project.dto.request.ProjectCreateRequest;
import com.buildflow.project.dto.request.ProjectUpdateRequest;
import com.buildflow.project.dto.response.ProjectResponse;
import com.buildflow.project.enums.ProjectStatus;

import java.util.List;

public interface ProjectService {

    ProjectResponse createProject(ProjectCreateRequest request);

    ProjectResponse getProjectById(Long id);

    List<ProjectResponse> getAllProjects();

    ProjectResponse updateProject(Long id, ProjectUpdateRequest request);

    void deleteProject(Long id);

    ProjectResponse updateProjectStatus(Long id, ProjectStatus status);
}

File: frontend/src/services/projectService.ts
import { axiosClient } from '../api/axiosClient';
import type { Project, CreateProjectRequest } from '../types/project';

export const ProjectService = {
  getAllProjects: async (): Promise<Project[]> => {
    const response = await axiosClient.get<Project[]>('/projects');
    return response.data;
  },

  getActiveProjects: async (): Promise<Project[]> => {
    const response = await axiosClient.get<Project[]>('/projects/active');
    return response.data;
  },

  getProjectById: async (id: number): Promise<Project> => {
    const response = await axiosClient.get<Project>(`/projects/${id}`);
    return response.data;
  },

  createProject: async (data: CreateProjectRequest): Promise<Project> => {
    const response = await axiosClient.post<Project>('/projects', data);
    return response.data;
  },

  updateProjectStatus: async (id: number, status: string): Promise<Project> => {
    const response = await axiosClient.patch<Project>(`/projects/${id}/status`, null, {
      params: { status }
    });
    return response.data;
  },

  deleteProject: async (id: number): Promise<void> => {
    await axiosClient.delete(`/projects/${id}`);
  }
};
